### PR TITLE
Scheduler tracing

### PR DESCRIPTION
**Pull request by danielnorberg**

### CODE REVIEW

**Comment by NatashaL** [comment]
@danielnorberg How did you choose which codepaths to trace, or what information exactly are you looking to find with this tracing? so I now what to look for myself

**Comment by danielnorberg** [comment]
@NatashaL I was primarily looking to trace the `Scheduler.tick()`.

File: styx-scheduler-service/src/main/java/com/spotify/styx/StyxScheduler.java [hunk 1]
```diff
@@ -400,7 +404,7 @@ public void create(Environment environment) {
     startRuntimeConfigUpdate(styxConfig, tickExecutor, dequeueRateLimiter);
     startCleaner(cleaner, tickExecutor);
 
-    setupMetrics(stateManager, workflowCache, storage, dequeueRateLimiter, stats);
+    setupMetrics(queuedStateManager, workflowCache, storage, dequeueRateLimiter, stats);
```
Comment (NatashaL) [inline styx-scheduler-service/src/main/java/com/spotify/styx/StyxScheduler.java]: With this we'll miss tracing of calls made to QueuedStateManager for getting metrics. Is that intended? Can we send the proxy instead?

Comment (danielnorberg) [reply styx-scheduler-service/src/main/java/com/spotify/styx/StyxScheduler.java]: The `setupMetrics` method needs to access the `QueuedStateManager` directly, it cannot use the `StateManager` interface.

Comment (NatashaL) [reply styx-scheduler-service/src/main/java/com/spotify/styx/StyxScheduler.java]: 👍 sounds like it should be refactored. but that can go in another PR then

**Comment by NatashaL** [comment]
👍 to merge after tests pass

**Pull request merged**